src/components/AddItemForm.js

```diff
@@ -1,28 +1,75 @@
 import React from 'react';
-import { collection, addDoc } from 'firebase/firestore';
+import {
+  collection,
+  addDoc,
+  query,
+  where,
+  onSnapshot,
+  getDocs,
+} from 'firebase/firestore';
 import { db } from '../lib/firebase';
 import './AddItemForm.css';
 
 function AddItemForm() {
-  const submitItem = (event) => {
+  const normalizeValue = (value) => {
+    const punctuationRegex = /[^\w]/gi;
+    const emojiRegex = /[\u{1F600}-\u{1F64F}]/gu;
+    const normalizedValue = value
+      .toLowerCase()
+      .replace(punctuationRegex, '')
+      .replace(emojiRegex, '');
+    return normalizedValue;
+  };
+
+  const submitItem = async (event) => {
     event.preventDefault();
     const userToken = window.localStorage.getItem('userToken');
     const itemName = event.target.itemName.value;
+    const itemNameNormalize = normalizeValue(itemName);
     const purchaseInterval = event.target.nextPurchase.value;
     const lastPurchased = event.target.lastPurchased.value || null;
-    handleSubmission(itemName, purchaseInterval, userToken, lastPurchased);
-    event.target.reset();
-    alert('Item added!');
+    const checkItem = await isItemInDatabase(itemName, userToken);
+
+    if (!checkItem) {
+      handleSubmission(
+        itemName,
+        itemNameNormalize,
+        purchaseInterval,
+        userToken,
+        lastPurchased,
+      );
+      event.target.reset();
+      alert('Item added!');
+    } else {
+      alert('Item is already in your list');
+    }
+  };
+
+  const isItemInDatabase = async (itemName, userToken) => {
+    const q = query(
+      collection(db, 'list'),
+      where('userToken', '==', userToken),
+      where('itemNameNormalize', '==', normalizeValue(itemName)),
```

Comment:
Nitpick - you've already used `normalizeValue` on line 28. Could `isItemInDatabase` accept `itemNameNormalize` as a param?